airbyte-ci/connectors/metadata_service/lib/metadata_service/gcs_upload.py

```diff
@@ -153,7 +161,10 @@ def upload_metadata_to_gcs(bucket_name: str, metadata_file_path: Path, prereleas
     storage_client = storage.Client(credentials=credentials)
     bucket = storage_client.bucket(bucket_name)
 
-    if prerelease:
-        return _prerelease_upload(metadata, bucket, prerelease)
-    else:
-        return _latest_upload(metadata, bucket, metadata_file_path)
+    _icon_upload(metadata, bucket, metadata_file_path)
+
+    version_uploaded, version_blob_id = _version_upload(metadata, bucket, metadata_file_path)
+    if not prerelease:
+        latest_uploaded, _latest_blob_id = _latest_upload(metadata, bucket, metadata_file_path)
+
+    return version_uploaded or latest_uploaded, version_blob_id
```

Comment:
Will we ever see latest_uploaded here? in the case that we upload it, we'll have already uploaded the version too

Reply:
I.e. for a non prerelease, if version upload failed, we want to know about it even it latest_uploaded succeeded, and the other way around. I guess we need separate return statements for the 2 cases?

Reply:
Good callout.

It was bugging me to.

Updated to have a more robust return value